src/imageIOHandler/QtOIIOHandler.cpp

```diff
@@ -243,31 +287,74 @@ bool QtOIIOHandler::read(QImage *image)
+        }
     }
 
     // Shuffle channels to convert from OIIO to Qt
     else if(nchannels == 4)
     {
-        // qDebug() << "[QtOIIO] shuffle channels";
-        oiio::ImageSpec requestedSpec(inSpec.width, inSpec.height, nchannels, typeDesc);
-        oiio::ImageBuf tmpBuf(requestedSpec);
+        oiio::TypeDesc typeDesc = moreThan8Bits ? oiio::TypeDesc::UINT16 : oiio::TypeDesc::UINT8;
 
-        const std::vector<int> channelOrder = {2, 1, 0, 3}; // This one works, not sure why...
-        oiio::ImageBufAlgo::channels(tmpBuf, inBuf, 4, &channelOrder.front());
-        inBuf.swap(tmpBuf);
-        // qDebug() << "[QtOIIO] shuffle channels done";
-    }
+        if(moreThan8Bits) // same than: format == QImage::Format_RGBA64 || format == QImage::Format_RGBX64
+        {
+            qDebug() << "[QtOIIO] Convert '" << inSpec.format.c_str() << "'' OIIO image to 'uint16' Qt image.";
+#pragma omp parallel for
+            for(int y = 0; y < inSpec.height; ++y)
+            {
+                for(int x = 0; x < inSpec.width; ++x)
+                {
+                    float rgba[4] = {0.0, 0.0, 0.0, 1.0};
+                    inBuf.getpixel(x, y, rgba, 4);
 
-    // qDebug() << "[QtOIIO] create output QImage";
-    QImage result(inSpec.width, inSpec.height, format);
+                    quint64* p = (quint64*)(result.scanLine(y)) + x;
+                    QRgba64 color = QRgba64::fromRgba64(floatToUShort(rgba[0]), floatToUShort(rgba[1]), floatToUShort(rgba[2]), floatToUShort(rgba[3]));
+                    *p = (quint64)color;
+                }
+            }
+        }
+        else
+        {
+            oiio::ImageSpec requestedSpec(inSpec.width, inSpec.height, nchannels, typeDesc);
+            oiio::ImageBuf tmpBuf(requestedSpec);
+            // qDebug() << "[QtOIIO] shuffle channels";
 
-    {
-        oiio::ROI exportROI = inBuf.roi();
-        exportROI.chbegin = 0;
-        exportROI.chend = nchannels;
+            std::vector<int> channelOrder = {0, 1, 2, 3};
+            float channelValues[] = {1.f, 1.f, 1.f, 1.f};
+            if(format == QImage::Format_ARGB32)
+            {
+                // (0xAARRGGBB) => 3, 0, 1, 2 in reverse order
+                channelOrder = {2, 1, 0, 3};
+            }
+            else if(format == QImage::Format_RGB32)
+            {
+                channelOrder = {2, 1, 0, -1}; // not sure
+            }
+            else
+            {
+                qWarning() << "Unsupported format conversion.";
+            }
+            // qWarning() << "channelOrder: " << channelOrder[0] << ", " << channelOrder[1] << ", " << channelOrder[2] << ", " << channelOrder[3];
+            oiio::ImageBufAlgo::channels(tmpBuf, inBuf, 4, &channelOrder.front(), channelValues, {}, false);
+            inBuf.swap(tmpBuf);
+            // qDebug() << "[QtOIIO] shuffle channels done";
+
+            {
+                oiio::ROI exportROI = inBuf.roi();
```

Comment:
const

Reply:
Thanks for the review! I just made the various changes.